client/my-sites/patterns/components/pattern-library/index.tsx

```diff
@@ -111,7 +111,7 @@ export const PatternLibrary = ( {
 	const { category, searchTerm, isGridView, patternTypeFilter, referrer } = usePatternsContext();
 
 	const { data: categories = [] } = usePatternCategories( locale );
-	const { data: patterns = [] } = usePatterns( locale, category, {
+	const { data: patterns } = usePatterns( locale, category, {
```

Comment:
```suggestion
	const { data: patterns = [], isFetching: isFetchingPatterns } = usePatterns( locale, category, {
```
Defaulting to an empty array is quite convenient (and this change currently breaks type checking, so we need to make some kind of change here).

If we want to pass `undefined` to `PatternsPageViewTracker`, I'd consider looking at the `isFetching` prop returned from `usePatterns` instead.

Reply:
@fredrikekelund I will merge to my PR and will address it there, to keep everything in one place

Reply:
@wongasy Do we want to pass `undefined` to `PatternsPageViewTracker ` is length is `0`? Or it's ok to pass `0`?

Reply:
@fredrikekelund addressed [`b42ffd8` (#89120)](https://github.com/Automattic/wp-calypso/pull/89120/commits/b42ffd88e1ef4e99f334948769a82ff087eb8879)